src/Access/tests/gtest_access_rights_ops.cpp

```diff
@@ -1,8 +1,198 @@
 #include <gtest/gtest.h>
 #include <Access/AccessRights.h>
+#include <Access/AccessRights.cpp>
+#include <IO/WriteBufferFromString.h>
+
+#include <list>
 
 using namespace DB;
 
+TEST(AccessRights, Radix)
+{
+    AccessRights root;
+    root.grant(AccessType::SELECT, "team");
+    root.grant(AccessType::SELECT, "toast");
+    root.grant(AccessType::SELECT, "toaster");
+    root.grant(AccessType::INSERT, "toaster", "bread");
+    root.grant(AccessType::ALTER_ADD_COLUMN, "toaster", "bread", "jam");
+
+    root.grantWildcard(AccessType::CREATE_TABLE, "t");
+
+    WriteBufferFromOwnString out;
+    root.dumpTree(out);
+
+    ASSERT_EQ(out.str(),
+              "Tree():  level=GLOBAL_LEVEL, name=NULL, flags=USAGE, min_flags=USAGE, max_flags=SELECT, INSERT, ALTER ADD COLUMN, CREATE TABLE, wildcard_grant=false, num_children=1\n"
+              "Tree(): - level=DATABASE_LEVEL, name=t, flags=CREATE TABLE, min_flags=CREATE TABLE, max_flags=SELECT, INSERT, ALTER ADD COLUMN, CREATE TABLE, wildcard_grant=true, num_children=2\n"
+              "Tree(): -- level=DATABASE_LEVEL, name=eam, flags=CREATE TABLE, min_flags=CREATE TABLE, max_flags=SELECT, CREATE TABLE, wildcard_grant=false, num_children=1\n"
+              "Tree(): --- level=DATABASE_LEVEL, name=NULL, flags=SELECT, CREATE TABLE, min_flags=SELECT, CREATE TABLE, max_flags=SELECT, CREATE TABLE, wildcard_grant=false, num_children=0\n"
```

Comment:
Maybe it makes sense to make it dump only non-zero values? I.e. to write "wildcard" only it's set, and num_children if it isn't zero, and name if it's not NULL?

Reply:
This is important info to verify in tests that radix compression works well and the tree structure is correct.

Reply:
It's just debug output. Ok, if it seems more useful to you to keep all the fields.